waku/v2/node/wakunode2.nim

```diff
@@ -33,6 +34,21 @@ type
   Topic* = waku_types.Topic
   Message* = seq[byte]
 
+  # NOTE based on Eth2Node in NBC eth2_network.nim
+  WakuNode* = ref object of RootObj
```

Comment:
This move was needed to get around recursive imports. There are other ways of addressing it, but this seemed like the most minimal one for now. If there's need we can pull out wakunode2 into its own folder/module and have types there.

In general, it seems like we put a bit too much stuff in node folder right now.